### PR TITLE
Properly escape types in annotations for 2020.2

### LINKED ISSUE #2473: Status bar error message, angle brackets do no display
image

I noticed the status bar error messages when hovering over an error in the editor, do not display properly. The error message was meant to be `mismatched types [E0308]expected \`Result<(), Error>;\`, found \`()\`'

### LINKED ISSUE #5241: HTML-escaped error messages
Hello,
  I'm not sure this is the right place to report this, but I am having issues with error messages that contain special characters: they are escaped as html entities.

I see: *Error:(340, 21) mismatched types [E0308]expected `Simd&#60;[u8; 8]&#62;`, found `Simd&#60;[i32; 8]&#62;`*

Instead of *Error:(340, 21) mismatched types [E0308]expected `Simd<[u8; 8]>;`, found `Simd<[i32; 8]>`*

## Environment

* **IntelliJ Rust plugin version:** 0.2.119.2201-201
* **Rust toolchain version:** 1.42.0 (b8cedc004 2020-03-09) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2020.1 Community Edition (IC-201.6668.121)
* **Operating system:** macOS 10.15.3

## Problem description

image

## Steps to reproduce

```rust
fn stbi_clamp_simd(x: i32x8) -> u8x8
{
    u8x8::from_cast(x.max(i32x8::splat(0)).min(i32x8::splat(255)))
}
```

### LINKED ISSUE #5598: E0308 warning popup doesn't display <> signs properly
## Environment

* **IntelliJ Rust plugin version:** 0.2.125.SNAPSHOT-202-dev
* **Rust toolchain version:** 1.44.0-nightly (b2e36e6c2 2020-04-22) x86_64-apple-darwin
* **IDE name and version:** CLion 2020.2 EAP (CL-202.5428.21)
* **Operating system:** macOS 10.15.4

## Problem description

Found the problem while checking #5548. `<` and `>` sings are displayed as `&#60;` and `&#62;`.

Screen Shot 2020-06-17 at 20 39 05

## Steps to reproduce

```rust
use std::collections::HashMap;

fn fib_dyn(n: u64, map: &mut HashMap<u64, u64>) -> u64 {
    match n {
        0 | 1 => 1,
        n => {
            if map.contains_key(&n) {
                *map.get(&n).unwrap()
            } else {
                let val = fib_dyn(n - 1, map) + fib_dyn(n - 2, map);
                map.insert(n, val);
                val
            }
        }
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn fib_5() {
        let mut map = HashMap::new();
        assert_eq!(fib_dyn(5, &map), 8);
    }
}
```

### PR DESCRIPTION
**Pull request by Undin**
Fixes #2473 (starting with 2020.2)
Fixes #5241 (starting with 2020.2)
Fixes #5598

### CODE REVIEW

**Comment by Undin** [comment]
bors retry

File: src/main/kotlin/org/rust/lang/utils/RsDiagnostic.kt [hunk 1]
```diff
@@ -1407,12 +1409,15 @@ private val RsSelfParameter.canonicalDecl: String
...
-private fun escapeTy(str: String): String = str
-    .replace("<", "&#60;")
-    .replace(">", "&#62;")
-    .replace("&", "&amp;")
+// BACKCOMPAT: 2020.1. Replace with `escapeString(str)`
+private fun escapeTy(str: String): String {
+    return if (ApplicationInfo.getInstance().build > BUILD_202) {
+        escapeString(str)
+    } else {
+        str.replace("<", "&#60;")
```
Comment (Stzx) [inline src/main/kotlin/org/rust/lang/utils/RsDiagnostic.kt:1417]: Oh, I didn't notice this PR last time. In fact, the problem is not that `escapeString(str)` needs to be added, but the original code replacement order is wrong.

```kotlin
str.replace("<", "&#60;").replace(">", "&#62;").replace("&", "&amp;")
```

Such a replacement order will cause & to be escaped incorrectly.

`<` -> `&#60;` -> `&amp;#60;`

Comment (Undin) [reply src/main/kotlin/org/rust/lang/utils/RsDiagnostic.kt:1417]: > In fact, the problem is not that escapeString(str) needs to be added, but the original code replacement order is wrong.

But it's not true. If you just change only order of `replace` calls, it'll fix only #5598 but not #2473 and #5241

image

**Pull request merged**